Source/CarthageKit/Errors.swift

```diff
@@ -89,6 +89,9 @@ public enum CarthageError: Error, Equatable {
 
 	/// An error occurred while shelling out.
 	case taskError(TaskError)
+	
+	/// An error occurred while generating the version file.
+	case versionFileError(description: String)
```

Comment:
`CarthageError` should really only be for user-visible failures.

Since caching is entirely optional, this error case should be removed. (If you need an error internally, you can add an `internal struct`.)